From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 31 Aug 2021 22:43:41 +0000
Subject: [PATCH] X11_KeyToUnicode

Fixed bug 1390 - X11_KeyToUnicode

manuel.montezelo 2012-01-22 08:56:18 PST

We had the following bug report at Debian:
http://bugs.debian.org/376560

Same one in Ubuntu:
https://bugs.launchpad.net/ubuntu/+source/libsdl1.2/+bug/66217

We've been including a patch since then (attached) to actually export the
symbol, since 2006.  In the last release the function seems to be there, alive
and kicking.

It's affecting other people too:
http://www.garagegames.com/community/forums/viewthread/52287
---
 src/video/x11/SDL_x11events.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/video/x11/SDL_x11events.c b/src/video/x11/SDL_x11events.c
index 85a71e7a..cfdbb157 100644
--- a/src/video/x11/SDL_x11events.c
+++ b/src/video/x11/SDL_x11events.c
@@ -1255,15 +1255,24 @@ static void get_modifier_masks(Display *display)
  * sequences (dead accents, compose key sequences) will not work since the
  * state has been irrevocably lost.
  */
+extern DECLSPEC Uint16 SDLCALL X11_KeyToUnicode(SDLKey, SDLMod);
+
 Uint16 X11_KeyToUnicode(SDLKey keysym, SDLMod modifiers)
 {
+	static int warning = 0;
 	struct SDL_VideoDevice *this = current_video;
 	char keybuf[32];
 	int i;
 	KeySym xsym = 0;
 	XKeyEvent xkey;
 	Uint16 unicode;
 
+	if ( warning ) {
+		warning = 0;
+		fprintf(stderr, "WARNING: Application is using X11_KeyToUnicode().\n");
+		fprintf(stderr, "This is not an official SDL function, please report this as a bug.\n");
+	}
+
 	if ( !this || !SDL_Display ) {
 		return 0;
 	}
